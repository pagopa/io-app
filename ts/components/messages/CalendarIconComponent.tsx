import * as React from "react";
import { G, Path, Svg, Text } from "react-native-svg";

type Props = {
  month: string;
  day: string;
  backgroundColor: string;
  textColor: string;
  small?: boolean;
};

class CalendarIconComponent extends React.PureComponent<Props, never> {
  get dimension(): number {
    return this.props.small ? 32 : 40;
  }

  public render() {
    const { month, day, backgroundColor, textColor } = this.props;
    return (
      <Svg height={this.dimension} width={this.dimension} viewBox="0 0 40 40">
        <G id="calendar">
          <Path
            d="m 33,0 h 3 c 2.209139,0 4,1.790861 4,4 v 32 c 0,2.209139 -1.790861,4 -4,4 H 4 C 1.790861,40 0,38.209139 0,36 V 4 C 0,1.790861 1.790861,0 4,0 h 3 v 3 h 4 V 0 h 18 v 3 h 4 z"
            id="icon"
            fill={backgroundColor}
          />
          <Text
            textAnchor="middle"
            alignmentBaseline="central"
            x="50%"
            y="30%"
            fill={textColor}
            fontSize={10}
          >
            {month}
          </Text>
          <Text
            textAnchor="middle"
            alignmentBaseline="central"
            x="50%"
            y="65%"
            fill={textColor}
            fontSize={18}
            fontWeight="bold"
          >
            {day}
          </Text>
        </G>
      </Svg>
    );
  }
}

export default CalendarIconComponent;
