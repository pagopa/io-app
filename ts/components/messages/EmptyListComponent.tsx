import { Text, View } from "native-base";
import * as React from "react";
import { Image, ImageSourcePropType, StyleSheet } from "react-native";
import customVariables from "../../theme/variables";

const styles = StyleSheet.create({
  view: {
    padding: customVariables.contentPadding,
    alignItems: "center"
  },
  title: {
    paddingTop: customVariables.contentPadding
  },
  subtitle: {
    textAlign: "center",
    paddingTop: customVariables.contentPadding,
    fontSize: customVariables.fontSizeSmall
  }
});

type Props = Readonly<{
  image: ImageSourcePropType;
  title: string;
  subtitle?: string;
}>;

export const EmptyListComponent = (props: Props) => {
  return (
    <View style={styles.view}>
      <View spacer={true} />
      <Image source={props.image} />
      <Text style={styles.title}>{props.title}</Text>
      {props.subtitle && <Text style={styles.subtitle}>{props.subtitle}</Text>}
    </View>
  );
};
