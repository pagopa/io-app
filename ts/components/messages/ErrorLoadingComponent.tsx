import I18n from "i18n-js";
import { Text, View } from "native-base";
import * as React from "react";
import { Image, StyleSheet } from "react-native";
import customVariables from "../../theme/variables";

const styles = StyleSheet.create({
  view: {
    padding: customVariables.contentPadding,
    alignItems: "center"
  },
  title: {
    paddingTop: customVariables.contentPadding
  }
});

export const ErrorLoadingComponent = () => {
  return (
    <View style={styles.view}>
      <View spacer={true} />
      <Image
        source={require("../../../img/messages/empty-message-list-icon.png")}
      />
      <Text style={styles.title}>{I18n.t("messages.loadingErrorTitle")}</Text>
    </View>
  );
};
