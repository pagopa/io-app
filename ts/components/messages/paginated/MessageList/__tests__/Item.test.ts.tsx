import React from "react";
import { fireEvent, render } from "@testing-library/react-native";

import { MessageCategory } from "../../../../../../definitions/backend/MessageCategory";
import { successReloadMessagesPayload } from "../../../../../__mocks__/messages";
import Item from "../Item";

jest.useFakeTimers();

const messages = successReloadMessagesPayload.messages;

const defaultProps: React.ComponentProps<typeof Item> = {
  category: { tag: "GENERIC" } as MessageCategory,
  hasPaidBadge: false,
  isRead: false,
  isArchived: false,
  isSelected: false,
  isSelectionModeEnabled: false,
  message: messages[0],
  onLongPress: jest.fn(),
  onPress: jest.fn()
};

describe("MessageList Item component", () => {
  describe("with all the flags set to false", () => {
    it("should match the snapshot", () => {
      const component = render(<Item {...defaultProps} />);
      expect(component.toJSON()).toMatchSnapshot();
    });
  });

  describe("when `hasPaidBadge` is true", () => {
    it("should match the snapshot", () => {
      expect(
        render(<Item {...defaultProps} hasPaidBadge={true} />).toJSON()
      ).toMatchSnapshot();
    });
  });

  describe("when `isRead` is true", () => {
    it("should match the snapshot", () => {
      expect(
        render(<Item {...defaultProps} isRead={true} />).toJSON()
      ).toMatchSnapshot();
    });
  });

  describe("when a EU_CODIV_CERT is present", () => {
    it("should match the snapshot", () => {
      expect(
        render(
          <Item
            {...defaultProps}
            category={{ tag: "EU_COVID_CERT" } as MessageCategory}
          />
        ).toJSON()
      ).toMatchSnapshot();
    });
  });

  describe("when `isSelectionModeEnabled` is true", () => {
    describe("and `isSelected` is false", () => {
      it("should match the snapshot", () => {
        expect(
          render(
            <Item {...defaultProps} isSelectionModeEnabled={true} />
          ).toJSON()
        ).toMatchSnapshot();
      });
    });
    describe("and `isSelected` is true", () => {
      it("should match the snapshot", () => {
        expect(
          render(
            <Item
              {...defaultProps}
              isSelectionModeEnabled={true}
              isSelected={true}
            />
          ).toJSON()
        ).toMatchSnapshot();
      });
    });
  });

  describe("when the user taps on the item", () => {
    it("should call only the `onPress` callback", () => {
      const onPress = jest.fn();
      const onLongPress = jest.fn();
      const component = render(
        <Item {...defaultProps} onPress={onPress} onLongPress={onLongPress} />
      );
      fireEvent(component.getByText(defaultProps.message.title), "onPress");
      expect(onPress).toHaveBeenCalledTimes(1);
      expect(onLongPress).not.toHaveBeenCalled();
    });
  });

  describe("when the user long-presses the item", () => {
    it("should call only the `onLongPress` callback", () => {
      const onPress = jest.fn();
      const onLongPress = jest.fn();
      const component = render(
        <Item {...defaultProps} onPress={onPress} onLongPress={onLongPress} />
      );
      fireEvent(component.getByText(defaultProps.message.title), "onLongPress");
      expect(onPress).not.toHaveBeenCalled();
      expect(onLongPress).toHaveBeenCalledTimes(1);
    });
  });
});
