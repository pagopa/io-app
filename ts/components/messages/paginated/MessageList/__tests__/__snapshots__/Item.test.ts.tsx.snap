// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList Item component when \`hasPaidBadge\` is true should match the snapshot 1`] = `
<View
  accessibilityLabel="Unread message, received by Ċentru tas-Saħħa, health. Għandek flus?. 
    received on October 18th 2021
  "
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "paddingVertical": 16,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      color="bluegreyDark"
      defaultColor="bluegreyDark"
      defaultWeight="SemiBold"
      font="TitilliumWeb"
      fontStyle={
        Object {
          "fontSize": 14,
        }
      }
      style={
        Array [
          Object {
            "fontSize": 14,
          },
          Object {
            "color": "#17324D",
            "fontFamily": "Titillium Web",
            "fontStyle": "normal",
            "fontWeight": "600",
          },
        ]
      }
      weight="SemiBold"
    >
      Ċentru tas-Saħħa
    </Text>
    <Text
      style={
        Object {
          "fontWeight": "bold",
          "lineHeight": 18,
          "marginBottom": -4,
        }
      }
    >
      18 Oct
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text>
      health
    </Text>
  </View>
  <View
    style={
      Object {
        "height": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "minHeight": 24,
        }
      }
    >
      <View
        style={
          Object {
            "alignSelf": "flex-start",
            "flex": 0,
            "paddingRight": 8,
            "paddingTop": 6.5,
          }
        }
      >
        <RNSVGSvgView
          bbHeight={10}
          bbWidth={10}
          focusable={false}
          height={10}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 10,
                "width": 10,
              },
            ]
          }
          width={10}
        >
          <RNSVGGroup>
            <RNSVGCircle
              cx="50%"
              cy="50%"
              fill={4278219750}
              propList={
                Array [
                  "fill",
                ]
              }
              r={5}
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <View
        style={
          Object {
            "width": "95%",
          }
        }
      >
        <Text
          color="bluegreyDark"
          font="TitilliumWeb"
          fontStyle={
            Object {
              "fontSize": 18,
            }
          }
          numberOfLines={2}
          style={
            Array [
              Object {
                "fontSize": 18,
              },
              Object {
                "color": "#17324D",
                "fontFamily": "Titillium Web",
                "fontStyle": "normal",
                "fontWeight": "600",
              },
            ]
          }
          weight="SemiBold"
          weightColorFactory={[Function]}
        >
          Għandek flus?
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "width": 90,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignSelf": "flex-start",
              "backgroundColor": "#ED1727",
              "borderRadius": 13.5,
              "height": 27,
              "justifyContent": "center",
              "paddingBottom": 3,
              "paddingLeft": 6,
              "paddingRight": 6,
              "paddingTop": 3,
            },
            Object {
              "borderStyle": "solid",
              "borderWidth": 1,
              "flexDirection": "row",
              "height": 25,
              "width": 65,
            },
            Object {
              "backgroundColor": "#00C5CA",
              "borderColor": "#00C5CA",
            },
          ]
        }
      >
        <Text
          color="bluegreyDark"
          defaultColor="bluegreyDark"
          defaultWeight="SemiBold"
          font="TitilliumWeb"
          fontStyle={
            Object {
              "fontSize": 14,
            }
          }
          style={
            Array [
              Object {
                "fontSize": 14,
              },
              Object {
                "color": "#17324D",
                "fontFamily": "Titillium Web",
                "fontStyle": "normal",
                "fontWeight": "600",
              },
            ]
          }
          weight="SemiBold"
        >
          Paid
        </Text>
      </View>
      <Text
        accessibilityElementsHidden={true}
        accessibilityLabel=""
        accessible={false}
        allowFontScaling={false}
        importantForAccessibility="no-hide-descendants"
        style={
          Array [
            Object {
              "color": "#0073E6",
              "fontSize": 24,
            },
            Array [
              Object {},
              Object {
                "color": "#0073E6",
              },
              Object {
                "fontSize": 24,
              },
            ],
            Object {
              "fontFamily": "io-icon-font",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        t
      </Text>
    </View>
  </View>
</View>
`;

exports[`MessageList Item component when \`isRead\` is true should match the snapshot 1`] = `
<View
  accessibilityLabel="Message, received by Ċentru tas-Saħħa, health. Għandek flus?. 
    received on October 18th 2021
  "
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "paddingVertical": 16,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      color="bluegreyDark"
      defaultColor="bluegreyDark"
      defaultWeight="SemiBold"
      font="TitilliumWeb"
      fontStyle={
        Object {
          "fontSize": 14,
        }
      }
      style={
        Array [
          Object {
            "fontSize": 14,
          },
          Object {
            "color": "#17324D",
            "fontFamily": "Titillium Web",
            "fontStyle": "normal",
            "fontWeight": "600",
          },
        ]
      }
      weight="SemiBold"
    >
      Ċentru tas-Saħħa
    </Text>
    <Text
      style={
        Object {
          "fontWeight": "bold",
          "lineHeight": 18,
          "marginBottom": -4,
        }
      }
    >
      18 Oct
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text>
      health
    </Text>
  </View>
  <View
    style={
      Object {
        "height": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "minHeight": 24,
        }
      }
    >
      <View
        style={
          Object {
            "width": "95%",
          }
        }
      >
        <Text
          color="bluegreyDark"
          font="TitilliumWeb"
          fontStyle={
            Object {
              "fontSize": 18,
            }
          }
          numberOfLines={2}
          style={
            Array [
              Object {
                "fontSize": 18,
              },
              Object {
                "color": "#17324D",
                "fontFamily": "Titillium Web",
                "fontStyle": "normal",
                "fontWeight": "600",
              },
            ]
          }
          weight="SemiBold"
          weightColorFactory={[Function]}
        >
          Għandek flus?
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "width": 90,
        }
      }
    >
      <Text
        accessibilityElementsHidden={true}
        accessibilityLabel=""
        accessible={false}
        allowFontScaling={false}
        importantForAccessibility="no-hide-descendants"
        style={
          Array [
            Object {
              "color": "#0073E6",
              "fontSize": 24,
            },
            Array [
              Object {},
              Object {
                "color": "#0073E6",
              },
              Object {
                "fontSize": 24,
              },
            ],
            Object {
              "fontFamily": "io-icon-font",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        t
      </Text>
    </View>
  </View>
</View>
`;

exports[`MessageList Item component when \`isSelectionModeEnabled\` is true and \`isSelected\` is false should match the snapshot 1`] = `
<View
  accessibilityLabel="Unread message, received by Ċentru tas-Saħħa, health. Għandek flus?. 
    received on October 18th 2021
  "
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "paddingVertical": 16,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      color="bluegreyDark"
      defaultColor="bluegreyDark"
      defaultWeight="SemiBold"
      font="TitilliumWeb"
      fontStyle={
        Object {
          "fontSize": 14,
        }
      }
      style={
        Array [
          Object {
            "fontSize": 14,
          },
          Object {
            "color": "#17324D",
            "fontFamily": "Titillium Web",
            "fontStyle": "normal",
            "fontWeight": "600",
          },
        ]
      }
      weight="SemiBold"
    >
      Ċentru tas-Saħħa
    </Text>
    <Text
      style={
        Object {
          "fontWeight": "bold",
          "lineHeight": 18,
          "marginBottom": -4,
        }
      }
    >
      18 Oct
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text>
      health
    </Text>
  </View>
  <View
    style={
      Object {
        "height": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "minHeight": 24,
        }
      }
    >
      <View
        style={
          Object {
            "alignSelf": "flex-start",
            "flex": 0,
            "paddingRight": 8,
            "paddingTop": 6.5,
          }
        }
      >
        <RNSVGSvgView
          bbHeight={10}
          bbWidth={10}
          focusable={false}
          height={10}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 10,
                "width": 10,
              },
            ]
          }
          width={10}
        >
          <RNSVGGroup>
            <RNSVGCircle
              cx="50%"
              cy="50%"
              fill={4278219750}
              propList={
                Array [
                  "fill",
                ]
              }
              r={5}
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <View
        style={
          Object {
            "width": "95%",
          }
        }
      >
        <Text
          color="bluegreyDark"
          font="TitilliumWeb"
          fontStyle={
            Object {
              "fontSize": 18,
            }
          }
          numberOfLines={2}
          style={
            Array [
              Object {
                "fontSize": 18,
              },
              Object {
                "color": "#17324D",
                "fontFamily": "Titillium Web",
                "fontStyle": "normal",
                "fontWeight": "600",
              },
            ]
          }
          weight="SemiBold"
          weightColorFactory={[Function]}
        >
          Għandek flus?
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "width": 90,
        }
      }
    >
      <Text
        accessibilityElementsHidden={true}
        accessibilityLabel=""
        accessible={false}
        allowFontScaling={false}
        importantForAccessibility="no-hide-descendants"
        style={
          Array [
            Object {
              "color": "#0073E6",
              "fontSize": 24,
            },
            Array [
              Object {},
              Object {
                "color": "#0073E6",
              },
              Object {
                "fontSize": 24,
              },
            ],
            Object {
              "fontFamily": "io-icon-font",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        R
      </Text>
    </View>
  </View>
</View>
`;

exports[`MessageList Item component when \`isSelectionModeEnabled\` is true and \`isSelected\` is true should match the snapshot 1`] = `
<View
  accessibilityLabel="Unread message, received by Ċentru tas-Saħħa, health. Għandek flus?. 
    received on October 18th 2021
  "
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "paddingVertical": 16,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      color="bluegreyDark"
      defaultColor="bluegreyDark"
      defaultWeight="SemiBold"
      font="TitilliumWeb"
      fontStyle={
        Object {
          "fontSize": 14,
        }
      }
      style={
        Array [
          Object {
            "fontSize": 14,
          },
          Object {
            "color": "#17324D",
            "fontFamily": "Titillium Web",
            "fontStyle": "normal",
            "fontWeight": "600",
          },
        ]
      }
      weight="SemiBold"
    >
      Ċentru tas-Saħħa
    </Text>
    <Text
      style={
        Object {
          "fontWeight": "bold",
          "lineHeight": 18,
          "marginBottom": -4,
        }
      }
    >
      18 Oct
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text>
      health
    </Text>
  </View>
  <View
    style={
      Object {
        "height": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "minHeight": 24,
        }
      }
    >
      <View
        style={
          Object {
            "alignSelf": "flex-start",
            "flex": 0,
            "paddingRight": 8,
            "paddingTop": 6.5,
          }
        }
      >
        <RNSVGSvgView
          bbHeight={10}
          bbWidth={10}
          focusable={false}
          height={10}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 10,
                "width": 10,
              },
            ]
          }
          width={10}
        >
          <RNSVGGroup>
            <RNSVGCircle
              cx="50%"
              cy="50%"
              fill={4278219750}
              propList={
                Array [
                  "fill",
                ]
              }
              r={5}
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <View
        style={
          Object {
            "width": "95%",
          }
        }
      >
        <Text
          color="bluegreyDark"
          font="TitilliumWeb"
          fontStyle={
            Object {
              "fontSize": 18,
            }
          }
          numberOfLines={2}
          style={
            Array [
              Object {
                "fontSize": 18,
              },
              Object {
                "color": "#17324D",
                "fontFamily": "Titillium Web",
                "fontStyle": "normal",
                "fontWeight": "600",
              },
            ]
          }
          weight="SemiBold"
          weightColorFactory={[Function]}
        >
          Għandek flus?
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "width": 90,
        }
      }
    >
      <Text
        accessibilityElementsHidden={true}
        accessibilityLabel=""
        accessible={false}
        allowFontScaling={false}
        importantForAccessibility="no-hide-descendants"
        style={
          Array [
            Object {
              "color": "#0073E6",
              "fontSize": 24,
            },
            Array [
              Object {},
              Object {
                "color": "#0073E6",
              },
              Object {
                "fontSize": 24,
              },
            ],
            Object {
              "fontFamily": "io-icon-font",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        S
      </Text>
    </View>
  </View>
</View>
`;

exports[`MessageList Item component when a EU_CODIV_CERT is present should match the snapshot 1`] = `
<View
  accessibilityLabel="Unread message, received by Ċentru tas-Saħħa, health. Għandek flus?. 
    received on October 18th 2021
  "
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "paddingVertical": 16,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      color="bluegreyDark"
      defaultColor="bluegreyDark"
      defaultWeight="SemiBold"
      font="TitilliumWeb"
      fontStyle={
        Object {
          "fontSize": 14,
        }
      }
      style={
        Array [
          Object {
            "fontSize": 14,
          },
          Object {
            "color": "#17324D",
            "fontFamily": "Titillium Web",
            "fontStyle": "normal",
            "fontWeight": "600",
          },
        ]
      }
      weight="SemiBold"
    >
      Ċentru tas-Saħħa
    </Text>
    <Text
      style={
        Object {
          "fontWeight": "bold",
          "lineHeight": 18,
          "marginBottom": -4,
        }
      }
    >
      18 Oct
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text>
      health
    </Text>
  </View>
  <View
    style={
      Object {
        "height": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "minHeight": 24,
        }
      }
    >
      <View
        style={
          Object {
            "alignSelf": "flex-start",
            "flex": 0,
            "paddingRight": 8,
            "paddingTop": 6.5,
          }
        }
      >
        <RNSVGSvgView
          bbHeight={10}
          bbWidth={10}
          focusable={false}
          height={10}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 10,
                "width": 10,
              },
            ]
          }
          width={10}
        >
          <RNSVGGroup>
            <RNSVGCircle
              cx="50%"
              cy="50%"
              fill={4278219750}
              propList={
                Array [
                  "fill",
                ]
              }
              r={5}
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <View
        style={
          Object {
            "width": "95%",
          }
        }
      >
        <Text
          color="bluegreyDark"
          font="TitilliumWeb"
          fontStyle={
            Object {
              "fontSize": 18,
            }
          }
          numberOfLines={2}
          style={
            Array [
              Object {
                "fontSize": 18,
              },
              Object {
                "color": "#17324D",
                "fontFamily": "Titillium Web",
                "fontStyle": "normal",
                "fontWeight": "600",
              },
            ]
          }
          weight="SemiBold"
          weightColorFactory={[Function]}
        >
          Għandek flus?
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "width": 90,
        }
      }
    >
      <View
        style={
          Object {
            "marginRight": 16,
          }
        }
      >
        <Text
          accessibilityElementsHidden={true}
          accessibilityLabel=""
          accessible={false}
          allowFontScaling={false}
          importantForAccessibility="no-hide-descendants"
          style={
            Array [
              Object {
                "color": "#0073E6",
                "fontSize": 12,
              },
              Array [
                Object {},
                Object {
                  "color": "#0073E6",
                },
                Object {},
              ],
              Object {
                "fontFamily": "io-icon-font",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          y
        </Text>
      </View>
      <Text
        accessibilityElementsHidden={true}
        accessibilityLabel=""
        accessible={false}
        allowFontScaling={false}
        importantForAccessibility="no-hide-descendants"
        style={
          Array [
            Object {
              "color": "#0073E6",
              "fontSize": 24,
            },
            Array [
              Object {},
              Object {
                "color": "#0073E6",
              },
              Object {
                "fontSize": 24,
              },
            ],
            Object {
              "fontFamily": "io-icon-font",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        t
      </Text>
    </View>
  </View>
</View>
`;

exports[`MessageList Item component with all the flags set to false should match the snapshot 1`] = `
<View
  accessibilityLabel="Unread message, received by Ċentru tas-Saħħa, health. Għandek flus?. 
    received on October 18th 2021
  "
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "paddingVertical": 16,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      color="bluegreyDark"
      defaultColor="bluegreyDark"
      defaultWeight="SemiBold"
      font="TitilliumWeb"
      fontStyle={
        Object {
          "fontSize": 14,
        }
      }
      style={
        Array [
          Object {
            "fontSize": 14,
          },
          Object {
            "color": "#17324D",
            "fontFamily": "Titillium Web",
            "fontStyle": "normal",
            "fontWeight": "600",
          },
        ]
      }
      weight="SemiBold"
    >
      Ċentru tas-Saħħa
    </Text>
    <Text
      style={
        Object {
          "fontWeight": "bold",
          "lineHeight": 18,
          "marginBottom": -4,
        }
      }
    >
      18 Oct
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text>
      health
    </Text>
  </View>
  <View
    style={
      Object {
        "height": 4,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "minHeight": 24,
        }
      }
    >
      <View
        style={
          Object {
            "alignSelf": "flex-start",
            "flex": 0,
            "paddingRight": 8,
            "paddingTop": 6.5,
          }
        }
      >
        <RNSVGSvgView
          bbHeight={10}
          bbWidth={10}
          focusable={false}
          height={10}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 10,
                "width": 10,
              },
            ]
          }
          width={10}
        >
          <RNSVGGroup>
            <RNSVGCircle
              cx="50%"
              cy="50%"
              fill={4278219750}
              propList={
                Array [
                  "fill",
                ]
              }
              r={5}
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <View
        style={
          Object {
            "width": "95%",
          }
        }
      >
        <Text
          color="bluegreyDark"
          font="TitilliumWeb"
          fontStyle={
            Object {
              "fontSize": 18,
            }
          }
          numberOfLines={2}
          style={
            Array [
              Object {
                "fontSize": 18,
              },
              Object {
                "color": "#17324D",
                "fontFamily": "Titillium Web",
                "fontStyle": "normal",
                "fontWeight": "600",
              },
            ]
          }
          weight="SemiBold"
          weightColorFactory={[Function]}
        >
          Għandek flus?
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "width": 90,
        }
      }
    >
      <Text
        accessibilityElementsHidden={true}
        accessibilityLabel=""
        accessible={false}
        allowFontScaling={false}
        importantForAccessibility="no-hide-descendants"
        style={
          Array [
            Object {
              "color": "#0073E6",
              "fontSize": 24,
            },
            Array [
              Object {},
              Object {
                "color": "#0073E6",
              },
              Object {
                "fontSize": 24,
              },
            ],
            Object {
              "fontFamily": "io-icon-font",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        t
      </Text>
    </View>
  </View>
</View>
`;
